herd/AArch64Sem.ml

```diff
@@ -955,26 +955,12 @@ module Make
                 set_afdb m in
             let add_setbits_af ipte m =
               add_setbits (is_zero ipte.af_v) "af:0" set_af m in
-            let setbits =
-              match dir with
-              | Dir.W ->
-                 if hd && updatedb then
-                   add_setbits_db ipte m
-                 else if ha then
-                   add_setbits_af ipte m
-                 else m
-              | Dir.R ->
-                  if hd && updatedb then
-                    (* The case of a failed CAS with no write, but with a db update *)
-                    M.altT (
-                      add_setbits_db ipte m
-                    )( (* no need to check ha, because hd implies ha *)
-                      add_setbits_af ipte m
-                    )
-                  else if ha then
-                    add_setbits_af ipte m
-                 else m in
-            setbits in
+            (* Write or failed CAS with no write, but with a db update *)
+            if hd && updatedb then
+              add_setbits_db ipte m
```

Comment:
would it be worth to add
`assert dir = Dir.W`
and later
`assert dir = Dir.R`?